src/Main.java

```diff
@@ -1,5 +1,117 @@
 public class Main {
     public static void main(String[] args) {
-        System.out.println("Домашнее задание");
+
+        System.out.println("homework5.");
+        //
+        {
+            int clientOS = 0;
+
+            if (clientOS == 0) {
+                System.out.println("Установите версию приложения для ios по ссылке");
+            } else if (clientOS == 1) {
+                System.out.println("Установите версию приложения для Android по ссылке");
+            } else {
+                System.out.println("Неизвестная операционная система");
+            }
+        }
+
+        //
+
+        {
+            int clientOS = 0;
+            int clientDeviceYear = 2015;
+
+            if (clientOS != 0 && clientOS != 1) {
+                System.out.println("Неизвестная операционная система");
+
+            } else {
+                String operationSystem;
+                if (clientOS == 0) {
+                    operationSystem = "ios";
+                } else {
+                    operationSystem = "Android";
+                }
+
+                String message;
+                if (clientDeviceYear < 2015) {
+                    message = "Установите облегченную версию приложения для " + operationSystem +
+                            " по ссылке";
+                } else {
+                    message = "Установите облегченную версию приложения для " + operationSystem +
+                            " по ссылке";
+                }
+                System.out.println(message);
+
+            }
+
+            //
+
+            int year = 2021;
+            boolean isLeap = false;
+            if (year % 4 == 0) {
+                if (year % 100 == 0) {
+                    if (year % 400 == 0) {
+                        isLeap = true;
+                    }
+                } else {
+                    isLeap = true;
+                }
+            }
```

Comment:
не нужно никогда писать вложенные условия - это плохая практика. Надо стараться всегда упрощать код, здесь должен быть всего 1 if и 1 else:
```suggestion
            isLeap = yser % 4 == 0 && year % 100 != 0 || year % 400 == 0;
```